app/code/Magento/Downloadable/Helper/File.php

```diff
@@ -57,32 +68,38 @@ public function __construct(
                 self::$_mimeTypes[$key] = $value;
             }
         }
+        $this->driver = $driver ?: ObjectManager::getInstance()
+            ->create(DriverInterface::class);
     }
 
     /**
      * Upload file from temporary folder.
+     *
      * @param string $tmpPath
-     * @param \Magento\MediaStorage\Model\File\Uploader $uploader
+     * @param Uploader $uploader
+     *
      * @return array
      */
-    public function uploadFromTmp($tmpPath, \Magento\MediaStorage\Model\File\Uploader $uploader)
+    public function uploadFromTmp($tmpPath, Uploader $uploader)
     {
         $uploader->setAllowRenameFiles(true);
         $uploader->setFilesDispersion(true);
         $absoluteTmpPath = $this->_mediaDirectory->getAbsolutePath($tmpPath);
-        $result = $uploader->save($absoluteTmpPath);
+        $result = $uploader->save($absoluteTmpPath) ?: [];
```

Comment:
it  is better to preserve existing behavior (on PHP7.4) and return `false` in case of failure
https://github.com/magento/magento2/blob/9f43821af62ab177e44b51b112ff8d32d9b82813/app/code/Magento/Downloadable/Controller/Adminhtml/Downloadable/File/Upload.php#L96-L100

Reply:
@xmav Thanks for your review. The class `\Magento\Downloadable\Helper\File` is market as @api and method `uploadFromTmp` return type array. What should I change for backward compatibility? Thanks.

Reply:
restored the previous flow